src/main/java/mendel/dbmanager/Storage.java

```diff
@@ -0,0 +1,126 @@
+package mendel.dbmanager;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+import mendel.discretetask.Deadline;
+import mendel.discretetask.Event;
+import mendel.discretetask.Task;
+import mendel.discretetask.Todo;
+import mendel.mendelexception.ServerError;
+import mendel.metacognition.TaskList;
+
+/**
+ * The Storage class handles the interaction with the file system to manage storage
+ * and retrieval of data from a database file. It manages the creation, loading,
+ * and updating of tasks within a given file path.
+ */
+public class Storage {
+    private final String filePath;
+    private final File db;
+
+    /**
+     * Constructs a Storage object associated with a specific file path. If the file does not exist,
+     * it attempts to create the file and its directories.
+     *
+     * @param filePath the path of the file to store the tasks.
+     * @throws ServerError if the file cannot be created due to an IO error.
+     */
+    public Storage(String filePath) throws ServerError {
+        this.filePath = filePath;
+        File db = new File(filePath);
+        this.db = db;
+        if (!db.exists()) {
+            try {
+                String[] dirLst = filePath.split("/");
+                new File(dirLst[0]).mkdir();
+                db.createNewFile();
+            } catch (IOException e) {
+                throw new ServerError("File cannot be created. Check data directory");
+            }
+        }
+    }
+
+    /**
+     * Copies data from the database file into the TaskList.
+     * Each line is parsed into the appropriate Task object (Todo, Deadline, or Event).
+     *
+     * @param taskStorage the TaskList object to load tasks into.
+     * @throws ServerError if the file is not found or if the task type in the file is unidentifiable.
```

Comment:
Perhaps the first letter (eg. 'the', 'if') could be capitalized as per the coding standard. Maybe the "the" could be omitted to make comments shorter without affecting readability.